### ISSUE TITLE
[Small bug] Unhandled exception

### ISSUE THREAD

**Issue by Nintorac** [facebookresearch/visdom#506]
**Bug Description**
Applying a keyboard interrupt does not halt execution of the program. This hardly ever happens and usually a subsequent interrupt will do the job. Sometimes (in the case the server is overloaded) the code takes a long time to exit the _send block and each subsequent interrupt just shows the same error but execution still continues

**Reproduction Steps**
Enter steps to reproduce the behavior:
1. Post new data to the server
2. KeyboardInterrupt while the client is blocked uploading new data to the server in the _send block
3. An exception is shown but execution does not halt (see additional context for exception)

**Expected behavior**
When a keyboard interrupt is received the execution stops completely

**Additional context**
Stack trace of python client when attempting to KeyboardInterrupt
```
Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 380, in _make_request
    httplib_response = conn.getresponse(buffering=True)
TypeError: getresponse() got an unexpected keyword argument 'buffering'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/visdom/__init__.py", line 446, in _send
    data=json.dumps(msg),
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/requests/api.py", line 112, in post
    return request('post', url, data=data, json=json, **kwargs)
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/requests/api.py", line 58, in request
    return session.request(method=method, url=url, **kwargs)
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/requests/sessions.py", line 508, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/requests/sessions.py", line 618, in send
    r = adapter.send(request, **kwargs)
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/requests/adapters.py", line 440, in send
    timeout=timeout
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 601, in urlopen
    chunked=chunked)
  File "/home/ubuntu/miniconda3/lib/python3.6/site-packages/urllib3/connectionpool.py", line 383, in _make_request
    httplib_response = conn.getresponse()
  File "/home/ubuntu/miniconda3/lib/python3.6/http/client.py", line 1331, in getresponse
    response.begin()
  File "/home/ubuntu/miniconda3/lib/python3.6/http/client.py", line 297, in begin
    version, status, reason = self._read_status()
  File "/home/ubuntu/miniconda3/lib/python3.6/http/client.py", line 258, in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
  File "/home/ubuntu/miniconda3/lib/python3.6/socket.py", line 586, in readinto
    return self._sock.recv_into(b)
KeyboardInterrupt
Without the incoming socket you cannot receive events from the server or register event handlers to your Visdom client.

Exception in user code:
------------------------------------------------------------

```

**Comment by JackUrb** [comment]
This can be solved by narrowing the scope of exceptions caught at the socket here: https://github.com/facebookresearch/visdom/blob/13f179d7acdb638c260779a53ad1f32195b24915/py/visdom/__init__.py#L526

I'd be happy to accept a PR that replaces the `BaseException` with something that more appropriately captures the potential exceptions thrown `self.session.post`.

**Comment by JackUrb** [comment]
closed in db27b34
